### ISSUE TITLE
Allow grouping of tasks

### ISSUE THREAD

**Issue by tuukkamustonen** [fabric/fabric#1315]
Instead of `fab -l` giving out something like:

```
setup                   ...
workon                  ...
clean                   ...
analyze                 ...
test                    ...
run                     ...
doc                     ...
release                 ...
create_env              ...
update_env              ...
delete_env              ...
func_test               ...
perf_test               ...
```

I would be nice to get:

```
Initialize:
  setup                   ...
  workon                  ...

 Dev:
  clean                   ...
  analyze                 ...
  test                    ...
  run                     ...

Doc:
  doc                     ...

Release:
  release                 ...

Deploy:
  create_env              ...
  update_env              ...
  delete_env              ...

Test:
  func_test               ...
  perf_test               ...
```

Namespaces are not a solution to this as:

1) they require also the namespace to be given when invoking tasks (here the tasks are just grouped, they are invoked as before)
2) I don't want to split out my tasks into several files, as my fabfile is already rather simple

Could this be supported through something like:

```
@task(group='Dev')
```

Not quite sure how this grouping and namespaces would mix?

**Comment by JensRantil** [comment]
Proposed new title: `Allow grouping when listing tasks`. Otherwise we risk mixing up this issue with #1772.

**Comment by bitprophet** [comment]
This is fixed if you upgrade Invoke to, I think 1.2? It's out in a stable release now either way, check pyinvoke.org. Thanks!
